### PR TITLE
[SPARK-42697][WEBUI] Fix /api/v1/applications to return total uptime instead of 0 for the duration field

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Fix /api/v1/applications to return total uptime instead of 0 for duration

### Why are the changes needed?

Fix REST API OneApplicationResource

### Does this PR introduce _any_ user-facing change?

yes, /api/v1/applications will return the total uptime instead of 0 for the duration

### How was this patch tested?

locally build and run

```json
[ {
  "id" : "local-1678183638394",
  "name" : "SparkSQL::10.221.102.180",
  "attempts" : [ {
    "startTime" : "2023-03-07T10:07:17.754GMT",
    "endTime" : "1969-12-31T23:59:59.999GMT",
    "lastUpdated" : "2023-03-07T10:07:17.754GMT",
    "duration" : 20317,
    "sparkUser" : "kentyao",
    "completed" : false,
    "appSparkVersion" : "3.5.0-SNAPSHOT",
    "startTimeEpoch" : 1678183637754,
    "endTimeEpoch" : -1,
    "lastUpdatedEpoch" : 1678183637754
  } ]
} ]
```

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/ui/SparkUI.scala [hunk 1]
```diff
@@ -167,7 +167,7 @@ private[spark] class SparkUI private (
         attemptId = None,
         startTime = new Date(startTime),
         endTime = new Date(-1),
-        duration = 0,
+        duration = System.currentTimeMillis() - startTime,
```
Comment (mridulm) [inline core/src/main/scala/org/apache/spark/ui/SparkUI.scala:170]: This wont work when used in context of history server

Comment (yaooqinn) [reply core/src/main/scala/org/apache/spark/ui/SparkUI.scala:170]: This is SparkUI only, not used by HistoryServer

**Comment by yaooqinn** [comment]
cc @HyukjinKwon @srowen @dongjoon-hyun ,thanks

File: core/src/main/scala/org/apache/spark/ui/SparkUI.scala (hunk 1 above)
Comment (srowen) [reply core/src/main/scala/org/apache/spark/ui/SparkUI.scala:170]: Seems OK, I just wonder why it was ever always 0?

Comment (yaooqinn) [reply core/src/main/scala/org/apache/spark/ui/SparkUI.scala:170]: https://github.com/apache/spark/pull/10648/files#diff-32f7bbe4f1ae694bd35f180c2ff918ed7ce5da42f37560bbd4d5ade6c7c2783eR116

It was initially merged with a giant PR, which was for HistoryServer. So I guess this line didn't get much attention

Comment (mridulm) [reply core/src/main/scala/org/apache/spark/ui/SparkUI.scala:170]: `SparkUI` is used from history server - not sure if this specific api is used though.

Comment (yaooqinn) [reply core/src/main/scala/org/apache/spark/ui/SparkUI.scala:170]: > `SparkUI` is used from history server

Only the method `create` from the companion object of `SparkUI` is referenced by history server. The classes `SparkUI` and `HistoryServer` both inherit from `WebUI`, they don't share functionality with each other.

Comment (mridulm) [reply core/src/main/scala/org/apache/spark/ui/SparkUI.scala:170]: Sounds good, thanks for confirming !

**Review by mridulm — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks @mridulm @srowen, merged to master/3.4/3.3/3.2

**Review by dongjoon-hyun — COMMENTED** [review]
+1, late LGTM. Thank you, @yaooqinn and all.

**Pull request closed without merging**